integration-tests/chopsticks/src/tests/xcm/limitedReserveTransferSpiritnetHydraDxV2.test.ts

```diff
@@ -0,0 +1,52 @@
+import { test } from 'vitest'
+import { sendTransaction, withExpect } from '@acala-network/chopsticks-testing'
+
+import * as SpiritnetConfig from '../../network/spiritnet.js'
+import * as HydraDxConfig from '../../network/hydraDx.js'
+import { KILT, keysAlice } from '../../utils.js'
+import { spiritnetContext, hydradxContext, getFreeBalanceSpiritnet, getFreeBalanceHydraDxKilt } from '../index.js'
+import { getAccountDestinationV2, getNativeAssetIdLocation, getSiblingDestination } from '../../network/utils.js'
+import { checkBalance, createBlock, hexAddress, setStorage } from '../utils.js'
+
+const KILT_ASSET_V2 = { V2: [getNativeAssetIdLocation(KILT)] }
+
+test('Limited Reserve V2 Transfers from Spiritnet Account Alice -> HydraDx', async ({ expect }) => {
+	const { checkEvents, checkSystemEvents } = withExpect(expect)
+
+	// Set storage
+	await setStorage(spiritnetContext, SpiritnetConfig.assignNativeTokensToAccount([keysAlice.address]))
+	await setStorage(hydradxContext, HydraDxConfig.assignNativeTokensToAccount([keysAlice.address]))
+
+	// check initial balance
+	await checkBalance(getFreeBalanceSpiritnet, SpiritnetConfig.hydraDxSovereignAccount, expect)
+	await checkBalance(getFreeBalanceHydraDxKilt, HydraDxConfig.omnipoolAccount, expect)
+
+	const omniPoolAddress = hexAddress(HydraDxConfig.omnipoolAccount)
+	const hydraDxDestination = { V2: getSiblingDestination(HydraDxConfig.paraId) }
+	const beneficiary = getAccountDestinationV2(omniPoolAddress)
+
+	const signedTx = spiritnetContext.api.tx.polkadotXcm
+		.limitedReserveTransferAssets(hydraDxDestination, beneficiary, KILT_ASSET_V2, 0, 'Unlimited')
+		.signAsync(keysAlice)
+
+	const events = await sendTransaction(signedTx)
+
+	// Order matters here, we need to create a block on the sender first
+	await createBlock(spiritnetContext)
+	await createBlock(hydradxContext)
+
+	// Check events sender
+	checkEvents(events, 'xcmpQueue').toMatchSnapshot('sender events xcm queue pallet')
+	checkEvents(events, 'polkadotXcm').toMatchSnapshot('sender events xcm pallet')
+	checkEvents(events, { section: 'balances', method: 'Withdraw' }).toMatchSnapshot('sender events Balances')
+
+	// Check events receiver
+	checkSystemEvents(hydradxContext, { section: 'currencies', method: 'Deposited' }).toMatchSnapshot(
+		'receiver events currencies'
+	)
+	checkSystemEvents(hydradxContext, 'xcmpQueue').toMatchSnapshot('receiver events xcmpQueue')
+
+	// check balance
+	await checkBalance(getFreeBalanceSpiritnet, SpiritnetConfig.hydraDxSovereignAccount, expect, KILT)
```

Comment:
Also the balance checks I would all group them together around the sender action. So instead of diving the tests by what we are testing, diving them by sender and receiver, since that's what we likely want to check. What do you think?

Reply:
I do not have a strong opinion on that. If you think, it makes the flow clearer, I am in: [e6e5f44](https://github.com/KILTprotocol/kilt-node/pull/614/commits/e6e5f4478ddc5254d35b846f22c393f2f22b56d9)